contracts/examples/erc1155/src/lib.rs

```diff
@@ -0,0 +1,387 @@
+		// so the reverting is handled automatically if one of the transfers fails
+		for i in 0..type_ids.len() {
+			let type_id = &type_ids[i];
+
+			if self.get_is_fungible(type_id) == true {
+				let amount = &values[i];
+
+				let balance = match self.get_balance_mapper(&from).get(type_id) {
+					Some(b) => b,
+					None => return sc_error!("Address has no tokens of that type"),
+				};
+
+				require!(amount > &0, "Must transfer more than 0");
+				require!(amount <= &balance, "Not enough balance for id");
+
+				self.decrease_balance(&from, &type_id, &amount);
+				self.increase_balance(&to, &type_id, &amount);
+			} else {
+				let token_id = &values[i];
+
+				require!(
+					self.get_token_owner(&type_id, &token_id) == from,
+					"_from_ is not the owner of the token"
+				);
+
+				let amount = BigUint::from(1u32);
+
+				self.decrease_balance(&from, &type_id, &amount);
+				self.increase_balance(&to, &type_id, &amount);
+
+				self.set_token_owner(&type_id, &token_id, &to);
+			}
+		}
+
+		// self.transfer_batch_event(&caller, &from, &to, ids, amounts);
+
+		if self.is_smart_contract_address(&to) {
+			// TODO: async-call
+		}
+
+		Ok(())
+	}
+
+	#[endpoint(setApprovalForAll)]
+	fn set_approved_for_all(&self, operator: Address, approved: bool) {
+		let caller = self.get_caller();
+
+		self.set_is_approved(&operator, &caller, approved);
+
+		// self.approval_for_all_event(&caller, &operator, approved);
+	}
+
+	// returns assigned id
+	#[endpoint(createToken)]
+	fn create_token(&self, uri: &[u8], initial_supply: BigUint, is_fungible: bool) -> BigUint {
+		let big_uint_one = BigUint::from(1u32);
+
+		let creator = self.get_caller();
+		let type_id = self.get_last_valid_type_id() + big_uint_one.clone();
+
+		self.get_balance_mapper(&creator)
+			.insert(type_id.clone(), initial_supply.clone());
+		self.set_token_type_creator(&type_id, &creator);
+		self.set_is_fungible(&type_id, is_fungible);
+
+		if !is_fungible {
+			self.set_owner_for_range(&type_id, &big_uint_one, &initial_supply, &creator);
+			self.set_last_valid_token_id_for_type(&type_id, &initial_supply);
+		}
+
+		self.set_last_valid_type_id(&type_id);
+		self.set_token_type_uri(&type_id, uri);
+
+		// self.transfer_single_event(&caller, &from, &to, &id, &amount);
+
+		// uri event
+
+		type_id
+	}
+
+	#[endpoint]
+	fn mint(&self, type_id: BigUint, amount: BigUint) -> SCResult<()> {
+		let creator = self.get_token_type_creator(&type_id);
+
+		require!(
+			self.get_caller() == creator,
+			"Only the token creator may mint more tokens"
+		);
+
+		self.increase_balance(&creator, &type_id, &amount);
+
+		if self.get_is_fungible(&type_id) == false {
+			let id_first = self.get_last_valid_token_id_for_type(&type_id);
+			let id_last = id_first.clone() + amount;
+
+			self.set_owner_for_range(&type_id, &id_first, &id_last, &creator);
+
+			self.set_last_valid_token_id_for_type(&type_id, &id_last);
+		}
+
+		// self.transfer_single_event(&caller, &from, &to, &id, &amount);
```

Comment:
lets have these calls. the even function inside can have 0 implementation for now.

Reply:
This isn't a function call, this is the event trigger, which as said above, I'll leave them commented until an update is done for them.